test/invoice_repository_test.rb

```diff
@@ -3,14 +3,24 @@
 require './lib/helper'
 
 RSpec.describe InvoiceRepository do
-  let!(:sales_engine) {SalesEngine.from_csv({:invoices => "./data/invoices.csv"})}
-  let!(:invoice_repo) {sales_engine.invoices}
+  # let!(:sales_engine) {SalesEngine.from_csv({:invoices => "./data/invoices.csv"})}
```

Comment:
I had to re-do this test to adhere to the provided interaction patter. The `InvoiceRepository` class (as well as the other `Repository` classes) is not instantiated directly by calling `.new(*some_args)` - it is instantiated *through* the `SalesEngine.from_csv` method. You'll see what i mean in the version i just pushed. It's confusing, but it works now and will continue to work for future classes as long as we stick to this pattern.